src/decoder/stream.rs

```diff
@@ -596,9 +609,15 @@ impl StreamingDecoder {
                             self.current_chunk.crc.update(&type_str.0);
                         }
                         self.current_chunk.remaining = length;
-                        self.apng_seq_handled = false;
                         self.current_chunk.raw_bytes.clear();
-                        self.state = Some(ReadChunk(type_str));
+                        self.state = match type_str {
+                            chunk::fdAT => Some(U32(ApngSequenceNumber)),
```

Comment:
Thanks for the review!  For now I just wanted to acknowledge that I've seen the feedback.  I'll hopefully find some time tomorrow to tweak the PR tomorrow and to address this code review feedback.

> This should check that `length >= 4`. But since the old code doesn't handle reading sequence numbers broken across multiple fdAT chunks, so I think it is fine for us to return an unsupported error here for now.

Ooops.  I should have caught that my refactoring removed the error handling that used to happen in `buf.read_be()?`.  Thanks for catching this.

> Separately though, this code should really only be reading the sequence number for the first fdAT of each frame

[The `fdAT` section](https://wiki.mozilla.org/APNG_Specification#.60fdAT.60:_The_Frame_Data_Chunk) of the APNG spec seems to indicate that *every* `fdAT` chunk should start with the animation frame's `sequence_number`, but doesn't really explain how to handle 2 subsequent `fdAT` chunks that have a different `sequnce_number`.  And we can't use `libpng` as the reference implementation, because it is not aware of `fdAT` chunks nor APNG in general.  

I'll try to look into using testcases from https://philip.html5.org/tests/apng/tests.html which contains:
* "Basic split fdAT" and "Split fdAT with zero-length chunk" testcases
* "fdAT too small"

Reply:
Just created https://github.com/w3c/PNG-spec/issues/379 to seek clarification and hopefully have the next iteration of PNG be a little more clear about it

Reply:
Please go ahead and add a new `FormatErrorInner` variant. That type is intentionally not public so that we can add new variants without it being a breaking change

Reply:
> This should check that `length >= 4`. But since the old code doesn't handle reading sequence numbers broken across multiple fdAT chunks, so I think it is fine for us to return an unsupported error here for now.

This is done now: added unit tests + fixed how the product code works.

> Separately though, this code should really only be reading the sequence number for the first fdAT of each frame

This is also done: I've just added a new unit test (I think the product code behaves as expected - see the comments in the new test code for more details).